### PR TITLE
Add document on how to implement a concept exercise in Go

### LINKED ISSUE #161: [Go] How to implement a Go concept exercise
[This document](https://github.com/exercism/v3/blob/master/languages/go/docs/implementing-a-concept-exercise.md) needs a first implementation. It should be targeted at non-maintainers that want to help out on a one-off basis.

As a start I suggest to look at [C#'s implementation guide](https://github.com/exercism/v3/blob/master/languages/csharp/docs/implementing-a-concept-exercise.md) and adjust that to Go's needs. From there it can be improved as we as maintainers get a better understanding of the process ourselves.

As a base we can copy this:  https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md

### PR DESCRIPTION
**Pull request by tehsphinx**
Closes #161

### CODE REVIEW

**Comment by tehsphinx** [comment]
@ErikSchierboom Could you have a look at this?

File: languages/go/docs/style-guide.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Style Guide
+
+## Stub functions
+
+Stub functions best contain a `panic` containing the phrase: `Please implement the XY function`.
```
Comment (DavyJ0nes) [inline languages/go/docs/style-guide.md]: nitpick word change :) 
```suggestion
Stub functions should implement a `panic` call, containing the phrase: `please implement the XY function`.
```

File: languages/go/docs/style-guide.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+Example:
+```go
+// Message extracts the message from the provided log line.
+func Message(line string) string {
+	panic("Please implement the Message function")
```
Comment (DavyJ0nes) [inline languages/go/docs/style-guide.md]: massive nitpick from me, but should panic messages start lower case, as with [error messages](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)?

Comment (tehsphinx) [reply languages/go/docs/style-guide.md]: Good question... I would have done the same as you suggest. They were capitalised in the experiment: https://github.com/exercism/research_experiment_1/blob/master/exercises/go-1-a/go_1_a.go#L5

I think better remove the capitalising!

**Comment by tehsphinx** [comment]
@DavyJ0nes: Updated. Agree with you on the `no capitalization on error messages`. Thought the same when I saw that in the research-experiment-1 code.

**Comment by tehsphinx** [comment]
@ErikSchierboom feel free to comment on this if you want some changes. I merged so I can better link to the documents.

**Comment by ErikSchierboom** [comment]
@tehsphinx LGTM!

**Pull request merged**